tripy/tripy/frontend/trace/ops/slice.py

```diff
@@ -197,10 +198,14 @@ def convert_to_positive_idx(index: Union[int, Tensor]) -> Union[int, Tensor]:
         # when dealing with a slice (not a single index), we clamp the start and end bounds to [0, t_shape[i]]
         # because out of bounds indices for a *slice* mean that the dim should be empty, not an error
         def clamp_bound(bound: Union[int, Tensor]) -> Union[int, Tensor]:
+            if isinstance(bound, int) and bound < 0:
+                return 0
+
+            # the dimension needs to be a tensor for the condition to where
+            shape_dim = t_shape[i] if isinstance(t_shape[i], Tensor) else DimensionSize(t_shape[i])
             if isinstance(bound, int):
-                return 0 if bound < 0 else where(bound > t_shape[i], t_shape[i], Tensor([bound]))
-            else:
-                return where(bound < 0, Tensor([0]), where(bound > t_shape[i], t_shape[i], bound))
+                return where(bound > shape_dim, shape_dim, Tensor([bound]))
+            return where(bound < 0, Tensor([0]), where(bound > shape_dim, shape_dim, bound))
```

Comment:
Not related to your change, but why did we use `where` instead of `min`/`max`?

Reply:
I don't think there's a reason; I'll replace them.